src/Editor/EditorSupport.php

```diff
@@ -44,27 +44,7 @@ public function register() {
 	 * Shows a notice in the editor if the Gutenberg or WP version prevents plugin features from working.
 	 */
 	public function maybe_show_notice() {
-		$screen = get_current_screen();
-
-		if ( ! $screen ) {
-			return;
-		}
-
-		$is_block_editor = (
-			! empty( $screen->is_block_editor )
-			||
-			// Applicable to Gutenberg v5.5.0 and older.
-			( function_exists( 'is_gutenberg_page' ) && is_gutenberg_page() )
-		);
-		if ( ! $is_block_editor ) {
-			return;
-		}
-
-		if ( ! in_array( get_post_type(), AMP_Post_Type_Support::get_eligible_post_types(), true ) ) {
-			return;
-		}
-
-		if ( $this->editor_supports_amp_block_editor_features() ) {
+		if ( $this->is_current_screen_supported_block_editor_for_amp_enabled_post_type() ) {
 			return;
 		}
```

Comment:
I just realized that this is currently resulting in the `wp_add_inline_script()` logic being run for all screens other than the edit post screen. Like if I go to the Dashboard, `is_current_screen_supported_block_editor_for_amp_enabled_post_type()` is returning `false` because it's not the block editor.

Maybe putting all that logic in the same method goes too far. 

Maybe instead of `is_current_screen_supported_block_editor_for_amp_enabled_post_type` we need a `is_block_editor_screen_for_amp_enabled_post_type` which looks like:

```php
public function is_block_editor_screen_for_amp_enabled_post_type() {
	$screen = get_current_screen();
	return (
		$screen
		&&
		! empty( $screen->is_block_editor )
		&&
		in_array( get_post_type(), AMP_Post_Type_Support::get_supported_post_types(), true )
	);
}
``` 

Then this method can instead be starting with:

```php
if ( $this->dependency_support->has_support() ) {
	return;
}

if ( ! $this->is_block_editor_screen_for_amp_enabled_post_type() ) {
	return;
}
```

Reply:
Oh right, nice catch! I've made the amendment in 5c93b0045a5f291fe32c8f048aac9c262c66291b.